src/main/java/taskmaster/storage/Storage.java

```diff
@@ -0,0 +1,139 @@
+package taskmaster.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import taskmaster.Deadline;
+import taskmaster.Event;
+import taskmaster.Task;
+import taskmaster.TaskType;
+import taskmaster.ToDo;
+
+import taskmaster.ui.Ui;
+
+/**
+ * Import Data from and
+ * Export Data to text file
+ *
+ * @author Emannuel Tan Jing Yue
+ * @since 2025-09-21
+ */
+public class Storage {
+    protected String filePathString;
+    protected File inputFile;
+    protected Path filePath;
+    protected Ui ui;
+
+    public Storage(String filePathString, Ui ui) {
+        this.filePathString = filePathString;
+        inputFile = new File(filePathString);
+        filePath = Paths.get(filePathString);
+        this.ui = ui;
+    }
+
+    /**
+     * Create directory and file if does not exist
+     */
+    private void createNewFile() {
+        try {
+            // Create data directory and TaskMaster.txt file
```

Comment:
what happens if directory path exists? only half the behavior is documented here.